From 9cf035b99a127a94f731f4447ce1040e4046e169 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 3 Feb 2020 10:03:19 +0100
Subject: [PATCH] evl/flag: clarify the pulse interface

Sending a pulse to a flag means flushing it with a broadcast signal,
so that all waiters unblock regardless of the actual value of that
flag. Make this obvious from the implementation.
---
 include/evl/flag.h | 29 ++++++++++++-----------------
 1 file changed, 12 insertions(+), 17 deletions(-)

diff --git a/include/evl/flag.h b/include/evl/flag.h
index 4a9ebed05174..752404bdf198 100644
--- a/include/evl/flag.h
+++ b/include/evl/flag.h
@@ -92,45 +92,40 @@ static inline void evl_raise_flag(struct evl_flag *wf)
 }
 
 /* wf->wait.lock held, irqs off */
-static inline void evl_pulse_flag_locked(struct evl_flag *wf)
+static inline void evl_flush_flag_locked(struct evl_flag *wf, int reason)
 {
-	evl_flush_wait_locked(&wf->wait, T_BCAST);
+	evl_flush_wait_locked(&wf->wait, reason);
 }
 
-static inline void evl_pulse_flag_nosched(struct evl_flag *wf)
+static inline void evl_flush_flag_nosched(struct evl_flag *wf, int reason)
 {
 	unsigned long flags;
 
 	evl_lock_flag(wf, flags);
-	evl_pulse_flag_locked(wf);
+	evl_flush_flag_locked(wf, reason);
 	evl_unlock_flag(wf, flags);
 }
 
-static inline void evl_pulse_flag(struct evl_flag *wf)
+static inline void evl_flush_flag(struct evl_flag *wf, int reason)
 {
-	evl_pulse_flag_nosched(wf);
+	evl_flush_flag_nosched(wf, reason);
 	evl_schedule();
 }
 
 /* wf->wait.lock held, irqs off */
-static inline void evl_flush_flag_locked(struct evl_flag *wf, int reason)
+static inline void evl_pulse_flag_locked(struct evl_flag *wf)
 {
-	evl_flush_wait_locked(&wf->wait, reason);
+	evl_flush_flag_locked(wf, T_BCAST);
 }
 
-static inline void evl_flush_flag_nosched(struct evl_flag *wf, int reason)
+static inline void evl_pulse_flag_nosched(struct evl_flag *wf)
 {
-	unsigned long flags;
-
-	evl_lock_flag(wf, flags);
-	evl_flush_flag_locked(wf, reason);
-	evl_unlock_flag(wf, flags);
+	evl_flush_flag_nosched(wf, T_BCAST);
 }
 
-static inline void evl_flush_flag(struct evl_flag *wf, int reason)
+static inline void evl_pulse_flag(struct evl_flag *wf)
 {
-	evl_flush_flag_nosched(wf, reason);
-	evl_schedule();
+	evl_flush_flag(wf, T_BCAST);
 }
 
 #endif /* _EVL_FLAG_H */
-- 
2.16.4

